### PR TITLE
Modify CLI and reference app to support global installation

### PR DESCRIPTION
**Pull request by ChuckFerring-zz**
No longer need to clone the CLI repo to use CLI and reference app.
Supports local and global installation.
Combined most of the helper classes to remove duplicated funtionality.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -3,21 +3,17 @@
...
     "colors": "^1.1.2",
     "commander": "^2.9.0",
+    "electron": "^1.6.2",
     "express": "^4.14.0",
-    "font-awesome": "^4.7.0",
     "glob": "^7.1.1",
     "inquirer": "^1.1.3",
-    "jquery": "^3.1.1",
     "open": "0.0.5",
-    "opent2t": "^1.0.1",
```
Comment (arjun-msft) [inline package.json:20]: don't you need this? #ByDesign

File: README.md [hunk 2]
```diff
@@ -39,7 +39,13 @@ npm install opent2t-translator-com-wink-thermostat
 ```
 2. First step is to do the onboarding. Wink communicates via the hub so you need to set that up first
 ```bash
-node index.js -o opent2t-translator-com-wink-hub -h WinkHub
+node_modules\.bin\opent2t-cli -o opent2t-translator-com-wink-hub -h WinkHub
+```
+
+Or, if installed globally
```
Comment (arjun-msft) [inline README.md:45]: dont you need to install the translators globally too? or can you install them in a folder and then run from that folder? #ByDesign

File: onboardingCli.js [hunk 3]
```diff
@@ -115,52 +238,6 @@ class OnboardingCli {
...
-
-    // this resolves variables in the onboarding flow with dynamic values retreived from the user
-    // takes in a key/value pair array and replces any key found in the string with the value in the array
-    // input
-    // my cool string with {key1} things to replace inside {key2} it
-    // {key1} : value1
-    // {key2} : value2
-    // output
-    // my cool string with value1 things to replace inside value2 it
-    replaceVarsInValue(value, replaceVars) {
```
Comment (arjun-msft) [inline onboardingCli.js:138]: why don't we need this anymore? #ByDesign

**Comment by arjun-msft** [comment]
:shipit:

File: package.json (hunk 1 above)
Comment (ChuckFerring-zz) [reply package.json:20]: No, it uses the opent2t module from the node_modules in the current directory.

---
In reply to: [105346247](https://github.com/openT2T/opent2t-cli/pull/28#discussion_r105346247) [](ancestors = 105346247)

File: README.md (hunk 2 above)
Comment (ChuckFerring-zz) [reply README.md:45]: It looks for the translator packages in the node_modules of the current directory.

---
In reply to: [105346389](https://github.com/openT2T/opent2t-cli/pull/28#discussion_r105346389) [](ancestors = 105346389)

File: onboardingCli.js (hunk 3 above)
Comment (ChuckFerring-zz) [reply onboardingCli.js:138]: I consolidated the helper methods in Opent2tHelper.js.

---
In reply to: [105346909](https://github.com/openT2T/opent2t-cli/pull/28#discussion_r105346909) [](ancestors = 105346909)

File: README.md [hunk 4]
```diff
@@ -17,16 +17,16 @@ Get your dev environment set up (PC or Mac):
 * [Install Node](https://nodejs.org/en/download/)
 * Choose your favorite IDE, e.g. [Visual Studio Code](https://code.visualstudio.com/).
 
-Clone the CLI repo
+Install the CLI module
 
 ```bash
-$ git clone https://github.com/openT2T/opent2t-cli.git
```
Comment (danielrischar) [inline README.md:23]: Can the CLI still be used without installing it as it is today?  If I npm install my translators in the opent2t-cli it should still work right?

**Comment by danielrischar** [comment]
:shipit:

**Comment by shalinijoshi19** [comment]
:shipit:

**Comment by danielrischar** [comment]
:shipit:

**Pull request merged**